apps/roam/src/components/CreateRelationDialog.tsx

```diff
@@ -0,0 +1,326 @@
+import React, { useState, useMemo } from "react";
+import { Dialog, Classes, Label, Button, Icon } from "@blueprintjs/core";
+import renderOverlay from "roamjs-components/util/renderOverlay";
+import { render as renderToast } from "roamjs-components/components/Toast";
+import MenuItemSelect from "roamjs-components/components/MenuItemSelect";
+import AutocompleteInput from "roamjs-components/components/AutocompleteInput";
+import getPageTitleByPageUid from "roamjs-components/queries/getPageTitleByPageUid";
+import getPageUidByPageTitle from "roamjs-components/queries/getPageUidByPageTitle";
+import getAllPageNames from "roamjs-components/queries/getAllPageNames";
+
+import { getSetting } from "~/utils/extensionSettings";
+import type { DiscourseNode } from "~/utils/getDiscourseNodes";
+import getDiscourseRelations, {
+  type DiscourseRelation,
+} from "~/utils/getDiscourseRelations";
+import { createReifiedRelation } from "~/utils/createReifiedBlock";
+import {
+  getDiscourseNodeTypeByTitle,
+  formatToRegexpText,
+} from "~/utils/getDiscourseNodeType";
+import getDiscourseNodes from "~/utils/getDiscourseNodes";
+
+export type CreateRelationDialogProps = {
+  onClose: () => void;
+  sourceNodeUid: string;
+};
+
+type RelWithDirection = DiscourseRelation & {
```

Comment:
Do we have existing types for this in discourse context or elsewhere to reuse?  Or a similar pattern but inline?  It seems like we should.  If so, let's make sure to share/reuse the type.